apps/server/src/AppService.ts

```
@@ -28,9 +31,39 @@ export class AppService {
     return this.ethersRpcProvider.getBalance(address);
   }
 
-  async getAllEventsFromBlockNumber(blockNumber: number): Promise<Event[]> {
+  async getAllEventsFromBlockNumber(): Promise<Event[]> {
     const currentBlockNumber = await this.ethersRpcProvider.getBlockNumber();
     const eventSignature = this.contract.filters.BRIDGE_TO_DEFI_CHAIN();
-    return this.contract.queryFilter(eventSignature, blockNumber, currentBlockNumber - 65);
+    const lastCheckedBlockNumber = await this.prisma.blockNumber.findFirst({
+      where: {
+        network: 'testnet',
+      },
+    });
+    // update last checked block number to currentBlockNumber - 64 if currentBlockNumber-64 > blockNumber in database
+    if (lastCheckedBlockNumber && currentBlockNumber - 64 > Number(lastCheckedBlockNumber.blockNumber)) {
```

`lastCheckedBlockNumber` should never be undefined here, and if it is, we should probably throw an error 🤔

---

Also, I'm thinking if this is a better way to check since it's more intuitive (bottom up instead of top down)

- If `lastCheckedBlockNumber + 64 > currentBlockNumber`
  - Exit early, there's nothing to check since there are not enough finalised blocks
- `lastCheckedBlockNumber + 64 < currentBlockNumber` (else)
  - There are 64 blocks between last checked and current, we `query` for those events)
  - We also update the `lastCheckedBlockNumber` to `lastCheckedBlockNumber + 64`

WDYT?

---

sounds good, lets do it